// @chakra-ui/styled-system/dist/cjs/config/transform.js
"use strict";

exports.__esModule = true;
exports.transform = void 0;

var _utils = require("@chakra-ui/utils");

var _utils2 = require("../utils");

var templates = {
  auto: "var(--chakra-transform)",
  "auto-gpu": "var(--chakra-transform-gpu)"
};

var degreeTransform = function degreeTransform(value) {
  if ((0, _utils.isCssVar)(value) || value == null) return value;
  return (0, _utils.isNumber)(value) ? value + "deg" : value;
};

var transform = {
  transform: {
    property: "transform",
    transform: function transform(value) {
      var _templates$value;

      return (_templates$value = templates[value]) != null ? _templates$value : value;
    }
  },
  transformOrigin: true,
  translateX: _utils2.t.spaceT("--chakra-translate-x"),
  translateY: _utils2.t.spaceT("--chakra-translate-y"),
  rotateX: {
    property: "--chakra-rotate-x",
    transform: degreeTransform
  },
  rotateY: {
    property: "--chakra-rotate-y",
    transform: degreeTransform
  },
  skewX: {
    property: "--chakra-skew-x",
    transform: degreeTransform
  },
  skewY: {
    property: "--chakra-skew-y",
    transform: degreeTransform
  }
};
exports.transform = transform;
